cmd/stack.go

```diff
@@ -352,21 +353,42 @@ func printStatus(cmd *cobra.Command, servicesStatus []stack.ServiceStatus) {
 	cmd.Println(t.Render())
 }
 
-func detectShell() (string, error) {
-	ppid := os.Getppid()
-
+func getParentInfo(ppid int) (types.ProcessInfo, error) {
 	parent, err := sysinfo.Process(ppid)
 	if err != nil {
-		return "", fmt.Errorf("cannot retrieve information for process %d: %w", ppid, err)
+		return types.ProcessInfo{}, fmt.Errorf("cannot retrieve information for process %d: %w", ppid, err)
 	}
 
 	parentInfo, err := parent.Info()
 	if err != nil {
-		return "", fmt.Errorf("cannot retrieve information for parent of process %d: %w", ppid, err)
+		return types.ProcessInfo{}, fmt.Errorf("cannot retrieve information for parent of process %d: %w", ppid, err)
 	}
 
-	shell := filepath.Base(parentInfo.Exe)
-	// NOTE: remove .exe extension from executable name present in Windows
+	return parentInfo, nil
+}
+
+func getShellName(exe string) string {
+	shell := filepath.Base(exe)
+	// NOTE: remove .exe extension from executable names present in Windows
 	shell = strings.TrimSuffix(shell, ".exe")
+	return shell
+}
+
+func detectShell() (string, error) {
+	ppid := os.Getppid()
+	parentInfo, err := getParentInfo(ppid)
+	if err != nil {
+		return "", err
+	}
+
+	shell := getShellName(parentInfo.Exe)
+	if shell == "go" {
```

Comment:
Nit. Please add a comment here about why we have this special case.

Reply:
What do you think of?
```suggestion
	// NOTE: handle the case of this CLI being run with "go run"
	// See https://github.com/elastic/elastic-package/pull/921
	if shell == "go" {
```

Reply:
Looks good, thanks.